### DISCUSSION TITLE
PythonOperator tasks are getting skipped in the DAG on Airflow 2.3.0

### DISCUSSION THREAD

**Discussion by vikasupendra** [apache/airflow#23902] [Q&A]
We recently moved to Airflow 2.3.0 on our Kubernetes Infrastructure and we have observed a few occurrences of tasks getting skipped leading to downstream tasks NOT being executed and the DAG fails. 

FYI - We have never observed this behavior on Airflow 2.2.x and by looking at the history of the kubernetes pod states and restarts for Airflow deployment, it seems that the service did not go down at any point in time.
Also, between the different DAG runs, none of the runtime parameters have changed except for small deltas in the data that is being processed in the pipeline.

By definition, PythonOperator tasks should NOT be skipped since conditional execution of tasks are associated with BranchPythonOperator.
I have not experimented with TriggerRule of 'all_success' or 'none_skipped'. Should I try to enforce a trigger rule for a PythonOperator as well? By default, PythonOperators have a trigger rule set to 'all_success' as per Airflow documentation.

Any pointers on why this could be happening and how we could make sure this does not happen?

![image](https://user-images.githubusercontent.com/16681656/170100014-0c5b6648-297f-454a-9278-4a77ebcedd15.png)

**Comment by potiuk** [comment]
I think you need to provide more details - logs describing why tasks have been skipped, definitions of your Dags etc. It's impossible to say what the problem was without investigation of conditions "around" the time it happened and providing us the logs. it's not a "known" issue.

**Reply by vikasupendra** [reply]
Thank you very much @potiuk for your response. Sure, I will work on getting the logs.

**Reply by Koojav** [reply]
@vikasupendra did you find the reason for this task behavior?

**Reply by Taragolis** [reply]
And also better to try upgrade airflow, if it possible, there is quite a lot changes since 2.3.0, see [Release Notes](https://airflow.apache.org/docs/apache-airflow/stable/release_notes.html)

**Reply by potiuk** [reply]
Yep. Absolutely agree with @Taragolis  - there are multiple hundreds of fixes already released in Airlfow since 2.3.0, upgrading Airflow should be thfirst thing to do in such cases.
